### PR TITLE
Add CIS Hardening Guide for 2.6x for RKE2 provisioned clusters

### PR DESCRIPTION
**Pull request by macedogm**
This PR add a hardening guide for RKE2 provisioned clusters in Rancher. It must be merged to master when RKE2 provision is GA.

Release notes is needed.

Close https://github.com/rancher/rancher/issues/36629.

### CODE REVIEW

**Comment by macedogm** [comment]
PR is ready, but marked as draft until QA can review it (CC @anupama2501).

**Comment by macedogm** [comment]
Release note comment:

**Synopsis:** Add RKE2 hardening guide for Rancher provisioned RKE2 clusters.
**Resolved (or not):** It will be resolved in 2.6.5 with GA of Rancher provisioned RKE2 clusters.
**Status:** PR is in draft waiting from QA validation in https://github.com/rancher/rancher/issues/36629
**Versions:** 2.6.5

File: scripts/converters/headers/header-2.6-rke2.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#### Testing controls methodology
+
+Rancher and RKE2 install Kubernetes services via Docker containers. Configuration is defined by arguments passed to the container at the time of initialization, not via configuration files.
+
+Where control audits differ from the original CIS benchmark, the audit commands specific to Rancher are provided for testing. When performing the tests, you will need access to the Docker command line on the hosts of all three RKE2 roles. The commands also make use of the [kubectl](https://kubernetes.io/docs/tasks/tools/) (with a valid configuration file) and [jq](https://stedolan.github.io/jq/) tools, which are required in the testing and evaluation of test results.
+
+> NOTE: Only `automated` tests (previously called `scored`) are covered in this guide.
+
+### Controls
+
```
Comment (btat) [inline scripts/converters/headers/header-2.6-rke2.md:35]: @macedogm Is there supposed to be content in this section?

Comment (macedogm) [reply scripts/converters/headers/header-2.6-rke2.md:35]: @btat Yes, but this is used as a template when we create the assessment pages, like this one for example - https://rancher.com/docs/rancher/v2.6/en/security/hardening-guides/1.6-benchmark-2.6/ .

Comment (btat) [reply scripts/converters/headers/header-2.6-rke2.md:35]: Gotcha 👍

File: content/rancher/v2.6/en/cis-scans/_index.md [hunk 2]
```diff
@@ -75,10 +75,15 @@ The following profiles are available:
...
 - RKE permissive 1.6
 - RKE hardened 1.6
+- RKE2 permissive 1.5
+- RKE2 hardened 1.5
+- RKE2 permissive 1.6
+- RKE2 hardened 1.6
+- K3s permissive 1.6
+- K3s hardened 1.6
+- AKS
 - EKS
```
Comment (anupama2501) [inline content/rancher/v2.6/en/cis-scans/_index.md:85]: @macedogm we haven't validated k3s completely as we have 1 more issue open. I think we should not include k3s here. Thoughts?

**Review by anupama2501 — CHANGES_REQUESTED** [review]

File: scripts/converters/headers/header-2.6-rke2.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+This document is to be used by Rancher operators, security teams, auditors and decision makers.
+
+For more detail about each audit, including rationales and remediations for failing tests, you can refer to the corresponding section of the CIS Kubernetes Benchmark v1.6. You can download the benchmark, after creating a free account, in [Center for Internet Security (CIS)](https://www.cisecurity.org/benchmark/kubernetes/).
+
+#### Testing controls methodology
+
+Rancher and RKE2 install Kubernetes services via Docker containers. Configuration is defined by arguments passed to the container at the time of initialization, not via configuration files.
+
+Where control audits differ from the original CIS benchmark, the audit commands specific to Rancher are provided for testing. When performing the tests, you will need access to the Docker command line on the hosts of all three RKE2 roles. The commands also make use of the [kubectl](https://kubernetes.io/docs/tasks/tools/) (with a valid configuration file) and [jq](https://stedolan.github.io/jq/) tools, which are required in the testing and evaluation of test results.
```
Comment (anupama2501) [inline scripts/converters/headers/header-2.6-rke2.md:30]: `you will need access to the Docker command line` would this be `command line`?

File: scripts/converters/headers/header-2.6-rke2.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Because Rancher and RKE2 install Kubernetes services as Docker containers, many of the control verification checks in the CIS Kubernetes Benchmark do not apply and will have a result of \`Not Applicable\`. This guide will walk through the various controls and provide updated example commands to audit compliance in Rancher created clusters. 
+
+This document is to be used by Rancher operators, security teams, auditors and decision makers.
+
+For more detail about each audit, including rationales and remediations for failing tests, you can refer to the corresponding section of the CIS Kubernetes Benchmark v1.6. You can download the benchmark, after creating a free account, in [Center for Internet Security (CIS)](https://www.cisecurity.org/benchmark/kubernetes/).
+
+#### Testing controls methodology
+
+Rancher and RKE2 install Kubernetes services via Docker containers. Configuration is defined by arguments passed to the container at the time of initialization, not via configuration files.
```
Comment (anupama2501) [inline scripts/converters/headers/header-2.6-rke2.md:28]: RKE2 uses containerd runtime

File: content/rancher/v2.6/en/security/hardening-guides/rke-1.6-hardening-2.6/_index.md [hunk 5]
```diff
@@ -26,7 +28,7 @@ This hardening guide is intended to be used for RKE clusters and associated with
 
 ### Overview
 
-This document provides prescriptive guidance for hardening a RKE cluster to be used for installing Rancher v2.6.3 with Kubernetes v1.18 up to v1.21 or provisioning a RKE cluster with Kubernetes v1.18 up to v.21 to be used within Rancher v2.6.3. It outlines the configurations required to address Kubernetes benchmark controls from the Center for Information Security (CIS).
+This document provides prescriptive guidance for hardening a RKE cluster to be used for installing Rancher v2.6 with Kubernetes v1.18 up to v1.21 or provisioning a RKE cluster with Kubernetes v1.18 up to v.21 to be used within Rancher v2.6. It outlines the configurations required to address Kubernetes benchmark controls from the Center for Information Security (CIS).
```
Comment (anupama2501) [inline content/rancher/v2.6/en/security/hardening-guides/rke-1.6-hardening-2.6/_index.md:31]: k8s up to 1.23

**Review by anupama2501 — CHANGES_REQUESTED** [review]

File: content/rancher/v2.6/en/cis-scans/_index.md (hunk 2 above)
Comment (macedogm) [reply content/rancher/v2.6/en/cis-scans/_index.md:85]: Agree. I'll remove the K3s reference and we update it with the other issue.

File: scripts/converters/headers/header-2.6-rke2.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+#### Overview
+
+This document is a companion to the [Rancher v2.6 RKE2 security hardening guide]({{<baseurl>}}/rancher/v2.6/en/security/hardening-guides/rke2-1.6-hardening-2.6/). The hardening guide provides prescriptive guidance for hardening a production installation of Rancher with RKE2 provisioned clusters, and this benchmark guide is meant to help you evaluate the level of security of the hardened cluster against each control in the benchmark.
+
+This guide corresponds to specific versions of the hardening guide, Rancher, CIS Benchmark and Kubernetes:
+
+| Hardening Guide Version | Rancher Version | CIS Benchmark Version |  Kubernetes Version |
+| ----------------------- | --------------- | --------------------- | ------------------- |
+| Hardening Guide CIS v1.6 Benchmark | Rancher v2.6.5 | CIS v1.6 | Kubernetes v1.21 and v1.22 |
```
[the remark this answers is not in the document]
Comment (macedogm) [reply scripts/converters/headers/header-2.6-rke2.md:18]: Fixed.

File: scripts/converters/headers/header-2.6-rke2.md (hunk 3 above)
Comment (macedogm) [reply scripts/converters/headers/header-2.6-rke2.md:30]: Yes, we can mention only `command line` here.

File: scripts/converters/headers/header-2.6-rke2.md (hunk 4 above)
Comment (macedogm) [reply scripts/converters/headers/header-2.6-rke2.md:28]: Fixed.

**Comment by macedogm** [comment]
@anupama2501 I updated the PR with your comments.

**Comment by macedogm** [comment]
PR https://github.com/rancherlabs/eio/issues/1016 to upload the hardening docs PDFs to Rancher's site.

File: content/rancher/v2.6/en/security/hardening-guides/_index.md [hunk 7]
```diff
@@ -36,27 +36,25 @@ These guides have been tested along with the Rancher v2.6 release. Each self-ass
...
 ### RKE2 Guides
 
-| Kubernetes Version | CIS Benchmark Version | Self Assessment Guide | Hardening Guides |
-| --- | --- | --- | --- |
-| Kubernetes v1.18 | CIS v1.5 | [Link](https://docs.rke2.io/security/cis_self_assessment15/) | [Link](https://docs.rke2.io/security/hardening_guide/) |
-| Kubernetes v1.20 | CIS v1.6 | [Link](https://docs.rke2.io/security/cis_self_assessment16/) | [Link](https://docs.rke2.io/security/hardening_guide/) |
+| Type | Kubernetes Version | CIS Benchmark Version | Self Assessment Guide | Hardening Guides |
+| ---- | ------------------ | --------------------- | --------------------- | ---------------- |
+| Rancher provisioned RKE2 cluster |  Kubernetes v1.21 up to v1.23 | CIS v1.6 | [Link]({{<baseurl>}}/rancher/v2.6/en/security/hardening-guides/rke2-1.6-benchmark-2.6) | [Link]({{<baseurl>}}/rancher/v2.6/en/security/hardening-guides/rke2-1.6-hardening-2.6) |
+| Standalone RKE2 | Kubernetes v1.21 and v1.22 | CIS v1.6 | [Link](https://docs.rke2.io/security/cis_self_assessment16/) | [Link](https://docs.rke2.io/security/hardening_guide/) |
```
Comment (anupama2501) [inline content/rancher/v2.6/en/security/hardening-guides/_index.md:52]: CIS scans should be applicable to v1.23 for standalone RKE2 as well here.

**Review by anupama2501 — CHANGES_REQUESTED** [review]

**Review by anupama2501 — APPROVED** [review]
lgtm

**Comment by btat** [comment]
@macedogm should I wait for the EIO ticket for file uploads to be done first before merging this PR?

**Comment by macedogm** [comment]
@btat the EIO issue is done now.

**Pull request merged**